dragonfly/block/crop.go

```diff
@@ -0,0 +1,22 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"math/rand"
+)
+
+type Crop interface {
+	// LightLevelRequired is the light level required for the crop to grow.
+	LightLevelRequired() uint8
+
+	// RequiresHydration is if the crop requires block Hydration in order to grow.
+	RequiresHydration() bool
+
+	// RequiresFarmland is if the crop requires farmland to be able to grow.
+	RequiresFarmland() bool
+
+	// Grow is ran every random tick to try and grow the crop 1 stage.
+	// Growth stages are handled inside of this method, making crops easier to create on a basis.
+	// LightLevel and Hydration are passed to make crops in better conditions grow faster.
+	Grow(pos world.BlockPos, w *world.World, r *rand.Rand, Hydration uint8)
```

Comment:
Are all of the methods here actually required? If the methods aren't needed outside of the implementation, you don't need to require the crops to implement all of these. Crops should really grow only when they themselves are ticked, not the farmland.

Reply:
I am ticking from the farmland because it requires a lot less to actually workout each of the methods. I already had to check to see if a crop was present on top of the farmland, so I decided that it'd be best if the crops grew in parity of farmland being ticked. It also makes for more accurate farmland states, which in hindsight aren't that important, but I feel like it adds more to it. I'd have to get the hydration of the farmland either way to make crops more effective to grow, so I settled on implementing it this way

Reply:
Might be better just follow vanilla behaviour and have them tick independently.